### PR TITLE
Enable unreachable_pub lint

### PR DESCRIPTION
**Pull request by bash**
Knowing which items are really public makes it easier to refactor those that are not public.

### CODE REVIEW

File: src/config/license.rs [hunk 1]
```diff
@@ -110,7 +110,7 @@ impl TemplateParser {
     /// "
     /// );
     /// ```
-    pub fn parse(template: &str) -> Result<String, LicenseError> {
+    pub(crate) fn parse(template: &str) -> Result<String, LicenseError> {
```
Comment (scampi) [inline src/config/license.rs]: General question: do you know if it compiles without `(crate)`, given that `TemplateParser` is already public for the crate only ?

Comment (bash) [reply src/config/license.rs]: The compiler complains if `parse` is `pub`:
```
warning: unreachable `pub` item
   --> src/config/license.rs:113:5
    |
113 |     pub fn parse(template: &str) -> Result<String, LicenseError> {
    |     ---^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |     |
    |     help: consider restricting its visibility: `pub(crate)`
    |
```

Comment (scampi) [reply src/config/license.rs]: nice thanks, love this.

**Comment by scampi** [comment]
LGTM

**Pull request merged**